File: leetcode-top-interview-150/169.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        major = None
        count = 0

        for num in nums:
            if count == 0:
                major = num
                count = 1
            elif major == num:
                count += 1
            else:
                count -= 1

        return major

File: leetcode-top-100-liked/169.kt
class Solution {
    fun majorityElement(nums: IntArray): Int {
        var major = nums[0]
        var count = 0

        for(num in nums) {
            if(count == 0) {
                major = num
                count = 1
            } else if(major == num) {
                count += 1
            } else {
                count -= 1
            }
        }
        return major
    }
}